File: frontend/src/services/NoteService.js
import axios from "axios";

const API = "http://localhost:8080/api/notes";

export const getAllNotes = () => axios.get(API);

export const searchNotes = (department, semester, subject) =>
  axios.get(`${API}/search`, {
    params: {
      department,
      semester,
      subject,
    },
  });

export const uploadNote = (formData) =>
  axios.post(`${API}/upload`, formData, {
    headers: {
      "Content-Type": "multipart/form-data",
    },
  });

export const getMyNotes = (email) =>
  axios.get(`${API}/my/${email}`);

export const deleteNote = (id) =>
  axios.delete(`${API}/${id}`);

export const updateNote = (id, note) =>
  axios.put(`${API}/${id}`, note);

File: backend/src/main/java/com/campusvault/service/NoteService.java
package com.campusvault.service;

import java.util.List;

import org.springframework.web.multipart.MultipartFile;

import com.campusvault.dto.NoteResponse;
import com.campusvault.entity.Note;

public interface NoteService {

    // Upload Note
    NoteResponse uploadNote(
            String title,
            String subject,
            String department,
            Integer semester,
            String uploadedBy,
            MultipartFile file) throws Exception;

    // Get All Notes
    List<Note> getAllNotes();

    // Search Notes
    List<Note> searchNotes(
            String department,
            Integer semester,
            String subject);

    // Get Notes Uploaded By User
    List<Note> getNotesByUploader(String email);

    // Update Note
    Note updateNote(Long id, Note note);

    // Delete Note
    void deleteNote(Long id);

}